cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/TimeScheduler.java

```diff
@@ -395,4 +436,31 @@ public Job newJob(TriggerFiredBundle bundle, org.quartz.Scheduler scheduler)
       }
     };
   }
+
+  /**
+   * @return Trigger key created from program, programType and scheduleName and TimeScheuler#PAUSED_NEW_TRIGGERS_GROUP
+   * if it exists in this group else returns the {@link TriggerKey} prepared with null which gets it with
+   * {@link Key#DEFAULT_GROUP}
+   * @throws org.quartz.SchedulerException
+   */
+  private TriggerKey getGroupedTriggerKey(Id.Program program, SchedulableProgramType programType, String scheduleName)
+    throws org.quartz.SchedulerException {
+
+    TriggerKey neverResumedTriggerKey = new TriggerKey(AbstractSchedulerService.scheduleIdFor(program, programType,
+                                                                                              scheduleName),
```

Comment:
this: new TriggerKey(AbstractSchedulerService.scheduleIdFor(program, programType, scheduleName), and this: new TriggerKey(AbstractSchedulerService.scheduleIdFor(program, programType, scheduleName)); appear to be repeated quite often. There also seems to be a bit of duplication between this method and line 285ff. Not sure if you could refactor that.